esphome/components/mqtt/mqtt_client.h

```diff
@@ -9,7 +9,11 @@
 #include "esphome/core/log.h"
 #include "esphome/components/json/json_util.h"
 #include "esphome/components/network/ip_address.h"
-#include <AsyncMqttClient.h>
+#ifdef USE_ESP_IDF
+#include "mqtt_backend_idf.h"
+#else
+#include "mqtt_backend_arduino.h"
+#endif
```

Comment:
Just for consistency
```suggestion
#if defined(USE_ESP_IDF)
#include "mqtt_backend_idf.h"
#elif defined(USE_ARDUINO)
#include "mqtt_backend_arduino.h"
#endif
```